Using ``finished`` on procs causes a codegen bug

### Example
```nim
proc p() = discard

echo p.finished
```
https://play.nim-lang.org/#ix=1K7v


### Current Output
```
Hint: used config file '/nim/config/nim.cfg' [Conf]
Hint: system [Processing]
Hint: widestrs [Processing]
Hint: io [Processing]
Hint: in [Processing]
Hint: gcc -c  -w  -I/nim/lib -I/usercode -o /usercode/nimcache/stdlib_io.nim.c.o /usercode/nimcache/stdlib_io.nim.c [Exec]
Hint: gcc -c  -w  -I/nim/lib -I/usercode -o /usercode/nimcache/stdlib_system.nim.c.o /usercode/nimcache/stdlib_system.nim.c [Exec]
Hint: gcc -c  -w  -I/nim/lib -I/usercode -o /usercode/nimcache/in.nim.c.o /usercode/nimcache/in.nim.c [Exec]
/usercode/nimcache/in.nim.c: In function 'finished__ALTXykuYNSMt9bxIBeb9c2Xwsystem':
/usercode/nimcache/in.nim.c:109:25: error: request for member 'ClE_0' in something not a structure or union
  109 |        result = ((NI*) x.ClE_0)[1] < 0;
      |                         ^
Error: execution of an external program failed: 'gcc -c  -w  -I/nim/lib -I/usercode -o /usercode/nimcache/in.nim.c.o /usercode/nimcache/in.nim.c'
```


### Expected Output
The error message should be clear that there isn't a proc called `finished` that takes a proc as a parameter.

### Additional Information
```
$ nim -v
Nim Compiler Version 1.0.0 [Linux: amd64]
Compiled at 2019-09-23
Copyright (c) 2006-2019 by Andreas Rumpf

git hash: f7a8fc46c0012033917582eb740dc0343c093e35
active boot switches: -d:release
```

--- a/lib/system.nim
+++ b/lib/system.nim
@@ -2412,10 +2412,13 @@ when notJSnotNims:
       {.error: "Only closure function and iterator are allowed!".}
 
   proc finished*[T: proc](x: T): bool {.noSideEffect, inline.} =
-    ## can be used to determine if a first class iterator has finished.
-    {.emit: """
-    `result` = ((NI*) `x`.ClE_0)[1] < 0;
-    """.}
+    ## It can be used to determine if a first class iterator has finished.
+    when T is "iterator":
+      {.emit: """
+      `result` = ((NI*) `x`.ClE_0)[1] < 0;
+      """.}
+    else:
+      {.error: "Only closure iterator is allowed!".}
 
 when defined(js):
   include "system/jssys"

--- a/tests/system/tsystem_misc.nim
+++ b/tests/system/tsystem_misc.nim
@@ -210,3 +210,10 @@ block: # Ordinal
   # doAssert enum is Ordinal # fails
   # doAssert Ordinal is SomeOrdinal
   # doAssert SomeOrdinal is Ordinal
+
+block:
+  proc p() = discard
+
+  doAssert not compiles(echo p.rawProc.repr)
+  doAssert not compiles(echo p.rawEnv.repr)
+  doAssert not compiles(echo p.finished)

